stdlib/public/core/Integers.swift.gyb

```diff
@@ -1678,7 +1678,82 @@ ${operatorComment(x.nonMaskingOperator, False)}
 
 }
 
-// Strideable conformance
+
+//===----------------------------------------------------------------------===//
+//===--- CustomStringConvertible conformance ------------------------------===//
+//===----------------------------------------------------------------------===//
+
+extension BinaryInteger {
+  @_inlineable // FIXME(sil-serialize-all)
+  @_versioned
+  @_transparent
+  internal func _description(
+    radix: Int = 10, uppercase: Bool = false
+  ) -> String {
+    _precondition(2...36 ~= radix, "Radix must be between 2 and 36")
+
+    if bitWidth <= 64 {
+      let radix_ = Int64(radix)
+      return Self.isSigned
+        ? _int64ToString(
+          Int64(truncatingIfNeeded: self), radix: radix_, uppercase: uppercase)
+        : _uint64ToString(
+          UInt64(truncatingIfNeeded: self), radix: radix_, uppercase: uppercase)
+    }
+
+    if self == (0 as Self) { return "0" }
+
+    // Bit shifting can be faster than division when `radix` is a power of two
+    // (although not necessarily the case for builtin types).
+    let isRadixPowerOfTwo = radix.nonzeroBitCount == 1
+    let radix_ = Magnitude(radix)
+    let quotientAndRemainder: (Magnitude) -> (Magnitude, Magnitude) = { value in
+      return isRadixPowerOfTwo
+        ? (value >> radix.trailingZeroBitCount, value & (radix_ - 1))
+        : value.quotientAndRemainder(dividingBy: radix_)
+    }
+
+    let hasLetters = radix > 10
+    let ascii: (UInt8) -> UInt8 = { digit in
```

Comment:
Can these be proper inner `func`s instead? That would help readability a little, imo.